docs/framework/chat-assistant.mdx

```diff
@@ -2,153 +2,325 @@
+
+    After adding the user's prompt to the conversational record, `generate_completion()` calls the `conversation` object's `stream_complete()` method, which generates an LLM completion based on the conversation so far. As its name implies, `stream_complete()` returns the completion as a stream of text chunks, which are captured and added to the `conversation` object.
   </Step>
 </Steps>
 
 
+## Bind the UI to the back-end code
+
+You've built the UI and written the code behind it. Let's connect the two! Go back to the browser window with the project editor and do the following:
+
+<Steps>
+  <Step title="Observe that the heading at the top of the app is now 'CHATBOT ASSISTANT'">
+  Earlier, you saw that the **Header** component's **Text** property was set to `@{my_app.title}`, a value in the app's `state` variable. You changed this value when you update the call to the Writer Framework's `init_state()` method.
+  </Step>
+
+  <Step title="Bind the Chatbot component to the 'state' variable's 'conversation' key">
+  Recall that the `conversation` object contained within the `state` variable contains the record of the conversation that the user is having with the LLM. Binding the **Chatbot** component to this object allows it to display the conversation to the user.
+
+  Select the **Chatbot** component. In the **Properties** panel, find the **Conversation** property and set its value to `@{conversation}`.
 
-## Binding to the UI
+  The value `@{conversation}` specifies that the **Chatbot** component should get its information from the value corresponding to the `conversation` key in the application's `state` variable.
+  </Step>
 
-Click on the **chatbot component** to open up the Component settings panel. Bind this chatbot to a conversation variable by adding `@{conversation}` in the Conversation Object property under General. This variable will reference the Writer AI SDK. You can also update properties such as the assistant's initials, the user's initials, and whether the chat uses markdown.
+  <Step title="Specify the Chatbot component's event handler">
+  You need to specify that the **Chatbot** component should call the `generate_completion()` function when the user enters a prompt. 
+  
+  Do this by scrolling down the **Properties** panel to the **Events** section until you see a property called **`wf_chatbot_message`**. Select **`generate_completion`** from its menu.
+  </Step>
+</Steps>
 
-![Finished chat assistant project](/framework/images/tutorial/chat/chat_assistant_5.png)
 
-Finally, attach the handler to the chatbot. In the User Interface, click on the **chatbot component** to bring up the Component settings panel. Scroll to the Events section towards the bottom of the pane and choose the `handle_simple_message` handler for the `wf-chatbot-message` event.
+## Run the application locally
 
-![Finished chat assistant project](/framework/images/tutorial/chat/chat_assistant_6.png)
+The application is ready to run locally!
```

Comment:
Let's make sure that it's clear somewhere in this tutorial and at this part that you can always see the running application using the "Preview" button at the top. As a reader I'd be wondering:

- Why do I have to quit and re-run it in order to see my changes?
- What's the difference between `writer edit` Preview and `writer run`

Let's answer those questions here.

Reply:
Updated.